src/modules/endpoint/Endpoint.js

```diff
@@ -0,0 +1,146 @@
+   * @return {object} The mapped data
+   */
+
+  /**
+   * @callback mapParamsToUrlCallback
+   * @param {object | undefined} state
+   * @return {string} The url
+   */
+
+  /**
+   * @param {string} name The name of this endpoint. This is used as key in the state and as Payload name. The Payload name is name + 'Paylaod'
+   * @param {function} mapParamsToUrl The mapper which maps the params to a request url
+   * @param {function} mapResponse Transforms the response from the fetch to a result
+   * @param responseOverride {*} An override value from the API response. Useful for testing.
+   * @param errorOverride {*} An override value to simulate an error while fetching. Useful for testing.
+   */
+  constructor (name, mapParamsToUrl, mapResponse, responseOverride, errorOverride) {
+    this.mapParamsToUrl = mapParamsToUrl
+    this.mapResponse = mapResponse
+    this.responseOverride = responseOverride
+    this.errorOverride = errorOverride
+    this._stateName = name
+    this.finishFetchAction = (payload: Payload) => createAction(finishFetchActionName(this._stateName))(payload)
+    this.startFetchAction = () => createAction(startFetchActionName(this._stateName))(new Payload(true))
+  }
+
+  /**
+   * @returns {string|*} The name of the linked state
+   */
+  get stateName () {
+    return this._stateName
+  }
+
+  async loadData (dispatch, oldPayload, params) {
+    const responseOverride = this.responseOverride
+    const errorOverride = this.errorOverride
+    /**
+     * Returns whether the correct data is available and ready for the fetcher to be displayed.
+     *
+     * @param state The state with (hopefully) all relevant params
+     * @param options The options get passed to the {@link mapResponse} function when fetching
+     * @return {function(*, *)} The Action for the redux store which can initiate a fetch
+     */
+    const formattedURL = this.mapParamsToUrl(params)
+
+    if (formattedURL.includes('undefined')) {
+      throw new Error(`Some necessary params in the state were undefined: ${formattedURL}`)
+    }
+
+    const lastUrl = oldPayload.requestUrl
+
+    if (lastUrl && lastUrl === formattedURL) {
+      // The correct data was already fetched
+      return oldPayload
+    }
+
+    // Fetch if the data is not valid anymore or it hasn't been fetched yet
+    dispatch(this.startFetchAction())
+
+    if (errorOverride) {
+      const payload = new Payload(false, null, errorOverride, formattedURL)
+      dispatch(this.finishFetchAction(payload))
+      return payload
+    }
+    if (responseOverride) {
+      const data = this.mapResponse(responseOverride, params)
+      const payload = new Payload(false, data, null, formattedURL)
+      dispatch(this.finishFetchAction(payload))
+      return payload
+    }
+
+    const payload = await this.fetchData(formattedURL, params)
+    dispatch(this.finishFetchAction(payload))
+    return payload
+  }
+
+  async fetchData (formattedUrl, params) {
+    return fetch(formattedUrl)
```

Comment:
should this function be async?
I think you could have it either non `async`, or have it `async` and do the following refactoring (I hope that works): 
```js
async fetchData (formattedUrl, params) {
  try {
    const response = await fetch(formattedUrl)
    if (!response.ok) {
      throw new Error(endpointLoadingErrorMessage)
    }
    const json = await response.json()
    const fetchedData = this.mapResponse(json, params)
    return Promise.resolve(new Payload(false, fetchedData, null, formattedUrl))
  } catch (e) {
    console.error(`Failed to fetch data for the endpoint: ${this.stateName}`)
    console.error(e)
    return new Payload(false, null, 'endpoint:page.loadingFailed', formattedUrl)
  }